src/pages/_app.jsx

```diff
@@ -45,16 +50,40 @@ const WrappedApp = ({ Component, pageProps }) => {
       if (httpError === 404) {
         return (
           <NotFoundPage
-            title={'Analysis doesn\'t exist.'}
+            title={'Analysis doesn\'t exist'}
             subTitle={'We searched, but we couldn\'t find the analysis you\'re looking for.'}
             hint='It may have been deleted by the project owner. Go home to see your own priejcts and analyses.'
           />
         );
       }
 
+      if (httpError === 403) {
+        return (
+          <NotFoundPage
+            title='Analysis not found'
+            subTitle={'You don\'t have access to this analysis. The owner may have made it private.'}
+            hint='If somebody gave you this link, they may need to invite you to their project.'
+          />
+        );
+      }
+
+      if (httpError === 401) {
+        return (
+          <NotFoundPage
+            title='Analysis not found'
+            subTitle={'You don\'t have access to this analysis.'}
+            hint='You may be able to view it by logging in.'
+          />
+        );
+      }
+
       return <Error statusCode={httpError} />;
     }
 
+    if (!amplifyConfigured) {
+      return <></>;
```

Comment:
Is this to manage a loading state?

Reply:
Correct. `_app.jsx` shouldn't load before the user is known to be authenticated. We load `_app.jsx` only when we successfully authenticate the user on the server side but the Amplify config needs to be applied on the browser as well. If we then load `ContentWrapper` before Amplify is configured we can end up sending pipeline status requests to an experiment with no JWT, therefore getting an Unauthorized message and an error on the UI.